lib/logstash/inputs/redis.rb

```diff
@@ -217,6 +246,86 @@ def list_runner(output_queue)
     end
   end
 
+  # private
+  def list_pattern_stop
+    return if @redis.nil? || !@redis.connected?
+
+    @redis.quit rescue nil
+    @redis = nil
+    @threadpool.shutdown
+    @threadpool.wait_for_termination
+  end
+
+  # private
+  def worker_consume(output_queue, key)
+    @logger.debug("Launched worker for #{key}")
+    redis = new_redis_instance
+    begin
+      (0..@pattern_list_max_items-1).each do
```

Comment:
There's fancy three dot version of the range operator that gives you the same range but excludes the last value (so should be identical to the manual `-1` here)